File: watchweb.py
# Python script to log the response code of a website
import sys
import time
import datetime
import requests

try:
    url = sys.argv[1]
    log = open('watchweb.log', 'a')
    while True:
        req = requests.get(url).status_code
        log.write('[%s] Response code on (%s) - %s' % (req, url, datetime.datetime.now()))
        time.sleep(300) # Sleep for 5 minutes
except KeyboardInterrupt:
    log.close()
    exit()

except IndexError:
    print("Usage: %s <url - https://example.com>" % sys.argv[0])
    
except:
    print("Unknown Error Occured")

File: watchweb.go
// Log the response code of a website
// Author = ScaredOS
// 4/15/2020
// ToDo:
//  Add more options:
//    Keyword check (Check for keyword, if not there, return error to log)\
//    Set Interval of check (Time to sleep for)
//    Set where to write the log to
package main
import (
  "fmt"
  "log"
  "net/http"
  "time"
  "os"
  "strconv"
)


func main() {
  url := os.Args[1]
  f, errr := os.Create("watchweb.log")
  if errr != nil {log.Fatal(errr)}
    for {
      resp, err := http.Get(url)
      if err != nil {log.Fatal(err)}
      stat := strconv.Itoa(resp.StatusCode)
      _, err = f.WriteString(fmt.Sprintf("[%s] Response code on (%s) - %s\n", stat, url, time.Now().Format("2006-01-02 15:04:05")))
      fmt.Printf("[%s] Response code on (%s) - %s\n", stat, url, time.Now().Format("2006-01-02 15:04:05"))
      if err != nil {log.Fatal(err)}
      time.Sleep(300 * time.Second)
    }
}
